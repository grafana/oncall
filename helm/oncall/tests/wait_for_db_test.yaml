suite: test init container wait-for-db in deployments
templates:
  - celery/deployment.yaml
  - engine/deployment.yaml
  - telegram-polling/deployment.yaml
release:
  name: oncall
chart:
  appVersion: v1.2.36
tests:
  - it: database.type=mysql -> should create initContainer for MySQL database (default)
    set:
      init.resources:
        limits:
          cpu: 100m
          memory: 128Mi
        requests:
          cpu: 100m
          memory: 128Mi
      telegramPolling.enabled: true
      database.type: mysql
    asserts:
      - contains:
          path: spec.template.spec.initContainers
          content:
            name: wait-for-db
          any: true
      - contains:
          path: spec.template.spec.initContainers[0].env
          content:
            name: MYSQL_DB_NAME
            value: oncall
      - matchSnapshot:
          path: spec.template.spec.initContainers

  - it: database.type=postgresql -> should create initContainer for PostgreSQL database
    set:
      telegramPolling.enabled: true
      database.type: postgresql
      externalPostgresql.host: some-postgresql-host
      init.resources:
        limits:
          cpu: 100m
          memory: 128Mi
        requests:
          cpu: 100m
          memory: 128Mi
    asserts:
      - contains:
          path: spec.template.spec.initContainers
          content:
            name: wait-for-db
          any: true
      - contains:
          path: spec.template.spec.initContainers[0].env
          content:
            name: DATABASE_TYPE
            value: postgresql
      - matchSnapshot:
          path: spec.template.spec.initContainers
