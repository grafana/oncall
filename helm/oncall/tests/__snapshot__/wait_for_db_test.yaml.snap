database.type=mysql -> should create initContainer for MySQL database (default):
  1: |
    - command:
        - sh
        - -c
        - until (python manage.py migrate --check); do echo Waiting for database migrations; sleep 2; done
      env:
        - name: BASE_URL
          value: https://example.com
        - name: SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: SECRET_KEY
              name: oncall
        - name: MIRAGE_SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: MIRAGE_SECRET_KEY
              name: oncall
        - name: MIRAGE_CIPHER_IV
          value: 1234567890abcdef
        - name: DJANGO_SETTINGS_MODULE
          value: settings.helm
        - name: AMIXR_DJANGO_ADMIN_PATH
          value: admin
        - name: OSS
          value: "True"
        - name: DETACHED_INTEGRATIONS_SERVER
          value: "False"
        - name: UWSGI_LISTEN
          value: "1024"
        - name: BROKER_TYPE
          value: rabbitmq
        - name: GRAFANA_API_URL
          value: http://oncall-grafana
        - name: MYSQL_HOST
          value: oncall-mariadb
        - name: MYSQL_PORT
          value: "3306"
        - name: MYSQL_DB_NAME
          value: oncall
        - name: MYSQL_USER
          value: root
        - name: MYSQL_PASSWORD
          valueFrom:
            secretKeyRef:
              key: mariadb-root-password
              name: oncall-mariadb
        - name: REDIS_PROTOCOL
          value: redis
        - name: REDIS_HOST
          value: oncall-redis-master
        - name: REDIS_PORT
          value: "6379"
        - name: REDIS_DATABASE
          value: "0"
        - name: REDIS_USERNAME
          value: ""
        - name: REDIS_PASSWORD
          valueFrom:
            secretKeyRef:
              key: redis-password
              name: oncall-redis
        - name: RABBITMQ_USERNAME
          value: user
        - name: RABBITMQ_PASSWORD
          valueFrom:
            secretKeyRef:
              key: rabbitmq-password
              name: oncall-rabbitmq
        - name: RABBITMQ_HOST
          value: oncall-rabbitmq
        - name: RABBITMQ_PORT
          value: "5672"
        - name: RABBITMQ_PROTOCOL
          value: amqp
        - name: RABBITMQ_VHOST
          value: ""
      image: grafana/oncall:v1.2.36
      imagePullPolicy: Always
      name: wait-for-db
      resources:
        limits:
          cpu: 100m
          memory: 128Mi
        requests:
          cpu: 100m
          memory: 128Mi
      securityContext: {}
  2: |
    - command:
        - sh
        - -c
        - until (python manage.py migrate --check); do echo Waiting for database migrations; sleep 2; done
      env:
        - name: BASE_URL
          value: https://example.com
        - name: SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: SECRET_KEY
              name: oncall
        - name: MIRAGE_SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: MIRAGE_SECRET_KEY
              name: oncall
        - name: MIRAGE_CIPHER_IV
          value: 1234567890abcdef
        - name: DJANGO_SETTINGS_MODULE
          value: settings.helm
        - name: AMIXR_DJANGO_ADMIN_PATH
          value: admin
        - name: OSS
          value: "True"
        - name: DETACHED_INTEGRATIONS_SERVER
          value: "False"
        - name: UWSGI_LISTEN
          value: "1024"
        - name: BROKER_TYPE
          value: rabbitmq
        - name: GRAFANA_API_URL
          value: http://oncall-grafana
        - name: MYSQL_HOST
          value: oncall-mariadb
        - name: MYSQL_PORT
          value: "3306"
        - name: MYSQL_DB_NAME
          value: oncall
        - name: MYSQL_USER
          value: root
        - name: MYSQL_PASSWORD
          valueFrom:
            secretKeyRef:
              key: mariadb-root-password
              name: oncall-mariadb
        - name: REDIS_PROTOCOL
          value: redis
        - name: REDIS_HOST
          value: oncall-redis-master
        - name: REDIS_PORT
          value: "6379"
        - name: REDIS_DATABASE
          value: "0"
        - name: REDIS_USERNAME
          value: ""
        - name: REDIS_PASSWORD
          valueFrom:
            secretKeyRef:
              key: redis-password
              name: oncall-redis
        - name: RABBITMQ_USERNAME
          value: user
        - name: RABBITMQ_PASSWORD
          valueFrom:
            secretKeyRef:
              key: rabbitmq-password
              name: oncall-rabbitmq
        - name: RABBITMQ_HOST
          value: oncall-rabbitmq
        - name: RABBITMQ_PORT
          value: "5672"
        - name: RABBITMQ_PROTOCOL
          value: amqp
        - name: RABBITMQ_VHOST
          value: ""
      image: grafana/oncall:v1.2.36
      imagePullPolicy: Always
      name: wait-for-db
      resources:
        limits:
          cpu: 100m
          memory: 128Mi
        requests:
          cpu: 100m
          memory: 128Mi
      securityContext: {}
  3: |
    - command:
        - sh
        - -c
        - until (python manage.py migrate --check); do echo Waiting for database migrations; sleep 2; done
      env:
        - name: BASE_URL
          value: https://example.com
        - name: SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: SECRET_KEY
              name: oncall
        - name: MIRAGE_SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: MIRAGE_SECRET_KEY
              name: oncall
        - name: MIRAGE_CIPHER_IV
          value: 1234567890abcdef
        - name: DJANGO_SETTINGS_MODULE
          value: settings.helm
        - name: AMIXR_DJANGO_ADMIN_PATH
          value: admin
        - name: OSS
          value: "True"
        - name: DETACHED_INTEGRATIONS_SERVER
          value: "False"
        - name: UWSGI_LISTEN
          value: "1024"
        - name: BROKER_TYPE
          value: rabbitmq
        - name: GRAFANA_API_URL
          value: http://oncall-grafana
        - name: MYSQL_HOST
          value: oncall-mariadb
        - name: MYSQL_PORT
          value: "3306"
        - name: MYSQL_DB_NAME
          value: oncall
        - name: MYSQL_USER
          value: root
        - name: MYSQL_PASSWORD
          valueFrom:
            secretKeyRef:
              key: mariadb-root-password
              name: oncall-mariadb
        - name: REDIS_PROTOCOL
          value: redis
        - name: REDIS_HOST
          value: oncall-redis-master
        - name: REDIS_PORT
          value: "6379"
        - name: REDIS_DATABASE
          value: "0"
        - name: REDIS_USERNAME
          value: ""
        - name: REDIS_PASSWORD
          valueFrom:
            secretKeyRef:
              key: redis-password
              name: oncall-redis
        - name: RABBITMQ_USERNAME
          value: user
        - name: RABBITMQ_PASSWORD
          valueFrom:
            secretKeyRef:
              key: rabbitmq-password
              name: oncall-rabbitmq
        - name: RABBITMQ_HOST
          value: oncall-rabbitmq
        - name: RABBITMQ_PORT
          value: "5672"
        - name: RABBITMQ_PROTOCOL
          value: amqp
        - name: RABBITMQ_VHOST
          value: ""
      image: grafana/oncall:v1.2.36
      imagePullPolicy: Always
      name: wait-for-db
      resources:
        limits:
          cpu: 100m
          memory: 128Mi
        requests:
          cpu: 100m
          memory: 128Mi
      securityContext: {}
database.type=postgresql -> should create initContainer for PostgreSQL database:
  1: |
    - command:
        - sh
        - -c
        - until (python manage.py migrate --check); do echo Waiting for database migrations; sleep 2; done
      env:
        - name: BASE_URL
          value: https://example.com
        - name: SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: SECRET_KEY
              name: oncall
        - name: MIRAGE_SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: MIRAGE_SECRET_KEY
              name: oncall
        - name: MIRAGE_CIPHER_IV
          value: 1234567890abcdef
        - name: DJANGO_SETTINGS_MODULE
          value: settings.helm
        - name: AMIXR_DJANGO_ADMIN_PATH
          value: admin
        - name: OSS
          value: "True"
        - name: DETACHED_INTEGRATIONS_SERVER
          value: "False"
        - name: UWSGI_LISTEN
          value: "1024"
        - name: BROKER_TYPE
          value: rabbitmq
        - name: GRAFANA_API_URL
          value: http://oncall-grafana
        - name: DATABASE_TYPE
          value: postgresql
        - name: DATABASE_HOST
          value: some-postgresql-host
        - name: DATABASE_PORT
          value: "5432"
        - name: DATABASE_NAME
          value: oncall
        - name: DATABASE_USER
          value: postgres
        - name: DATABASE_PASSWORD
          valueFrom:
            secretKeyRef:
              key: postgres-password
              name: oncall-postgresql-external
        - name: REDIS_PROTOCOL
          value: redis
        - name: REDIS_HOST
          value: oncall-redis-master
        - name: REDIS_PORT
          value: "6379"
        - name: REDIS_DATABASE
          value: "0"
        - name: REDIS_USERNAME
          value: ""
        - name: REDIS_PASSWORD
          valueFrom:
            secretKeyRef:
              key: redis-password
              name: oncall-redis
        - name: RABBITMQ_USERNAME
          value: user
        - name: RABBITMQ_PASSWORD
          valueFrom:
            secretKeyRef:
              key: rabbitmq-password
              name: oncall-rabbitmq
        - name: RABBITMQ_HOST
          value: oncall-rabbitmq
        - name: RABBITMQ_PORT
          value: "5672"
        - name: RABBITMQ_PROTOCOL
          value: amqp
        - name: RABBITMQ_VHOST
          value: ""
      image: grafana/oncall:v1.2.36
      imagePullPolicy: Always
      name: wait-for-db
      resources:
        limits:
          cpu: 100m
          memory: 128Mi
        requests:
          cpu: 100m
          memory: 128Mi
      securityContext: {}
  2: |
    - command:
        - sh
        - -c
        - until (python manage.py migrate --check); do echo Waiting for database migrations; sleep 2; done
      env:
        - name: BASE_URL
          value: https://example.com
        - name: SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: SECRET_KEY
              name: oncall
        - name: MIRAGE_SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: MIRAGE_SECRET_KEY
              name: oncall
        - name: MIRAGE_CIPHER_IV
          value: 1234567890abcdef
        - name: DJANGO_SETTINGS_MODULE
          value: settings.helm
        - name: AMIXR_DJANGO_ADMIN_PATH
          value: admin
        - name: OSS
          value: "True"
        - name: DETACHED_INTEGRATIONS_SERVER
          value: "False"
        - name: UWSGI_LISTEN
          value: "1024"
        - name: BROKER_TYPE
          value: rabbitmq
        - name: GRAFANA_API_URL
          value: http://oncall-grafana
        - name: DATABASE_TYPE
          value: postgresql
        - name: DATABASE_HOST
          value: some-postgresql-host
        - name: DATABASE_PORT
          value: "5432"
        - name: DATABASE_NAME
          value: oncall
        - name: DATABASE_USER
          value: postgres
        - name: DATABASE_PASSWORD
          valueFrom:
            secretKeyRef:
              key: postgres-password
              name: oncall-postgresql-external
        - name: REDIS_PROTOCOL
          value: redis
        - name: REDIS_HOST
          value: oncall-redis-master
        - name: REDIS_PORT
          value: "6379"
        - name: REDIS_DATABASE
          value: "0"
        - name: REDIS_USERNAME
          value: ""
        - name: REDIS_PASSWORD
          valueFrom:
            secretKeyRef:
              key: redis-password
              name: oncall-redis
        - name: RABBITMQ_USERNAME
          value: user
        - name: RABBITMQ_PASSWORD
          valueFrom:
            secretKeyRef:
              key: rabbitmq-password
              name: oncall-rabbitmq
        - name: RABBITMQ_HOST
          value: oncall-rabbitmq
        - name: RABBITMQ_PORT
          value: "5672"
        - name: RABBITMQ_PROTOCOL
          value: amqp
        - name: RABBITMQ_VHOST
          value: ""
      image: grafana/oncall:v1.2.36
      imagePullPolicy: Always
      name: wait-for-db
      resources:
        limits:
          cpu: 100m
          memory: 128Mi
        requests:
          cpu: 100m
          memory: 128Mi
      securityContext: {}
  3: |
    - command:
        - sh
        - -c
        - until (python manage.py migrate --check); do echo Waiting for database migrations; sleep 2; done
      env:
        - name: BASE_URL
          value: https://example.com
        - name: SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: SECRET_KEY
              name: oncall
        - name: MIRAGE_SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: MIRAGE_SECRET_KEY
              name: oncall
        - name: MIRAGE_CIPHER_IV
          value: 1234567890abcdef
        - name: DJANGO_SETTINGS_MODULE
          value: settings.helm
        - name: AMIXR_DJANGO_ADMIN_PATH
          value: admin
        - name: OSS
          value: "True"
        - name: DETACHED_INTEGRATIONS_SERVER
          value: "False"
        - name: UWSGI_LISTEN
          value: "1024"
        - name: BROKER_TYPE
          value: rabbitmq
        - name: GRAFANA_API_URL
          value: http://oncall-grafana
        - name: DATABASE_TYPE
          value: postgresql
        - name: DATABASE_HOST
          value: some-postgresql-host
        - name: DATABASE_PORT
          value: "5432"
        - name: DATABASE_NAME
          value: oncall
        - name: DATABASE_USER
          value: postgres
        - name: DATABASE_PASSWORD
          valueFrom:
            secretKeyRef:
              key: postgres-password
              name: oncall-postgresql-external
        - name: REDIS_PROTOCOL
          value: redis
        - name: REDIS_HOST
          value: oncall-redis-master
        - name: REDIS_PORT
          value: "6379"
        - name: REDIS_DATABASE
          value: "0"
        - name: REDIS_USERNAME
          value: ""
        - name: REDIS_PASSWORD
          valueFrom:
            secretKeyRef:
              key: redis-password
              name: oncall-redis
        - name: RABBITMQ_USERNAME
          value: user
        - name: RABBITMQ_PASSWORD
          valueFrom:
            secretKeyRef:
              key: rabbitmq-password
              name: oncall-rabbitmq
        - name: RABBITMQ_HOST
          value: oncall-rabbitmq
        - name: RABBITMQ_PORT
          value: "5672"
        - name: RABBITMQ_PROTOCOL
          value: amqp
        - name: RABBITMQ_VHOST
          value: ""
      image: grafana/oncall:v1.2.36
      imagePullPolicy: Always
      name: wait-for-db
      resources:
        limits:
          cpu: 100m
          memory: 128Mi
        requests:
          cpu: 100m
          memory: 128Mi
      securityContext: {}
